src/main/java/com/product/service/ProductService.java

```diff
@@ -0,0 +1,19 @@
+package com.product.service;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import com.product.model.Product;
+
+public interface ProductService {
+
+	public Collection<Product> getAllProducts();
```

Comment:
Why Collection (superclass of List).. It should be List instead

Read into Collection, List, ArrayList and whats the difference between all

Reply:
Done.